modules/document_repository/php/files.class.inc

```diff
@@ -312,109 +312,136 @@ class Files extends \NDB_Page
+                continue;
+            }
+
+            // Prepare fields for database entry once upload has completed.
+            $site = $body['forSite']       !== '' ? $body['forSite'] : null;
+            $site = $DB->pselectOne(
+                "SELECT CenterID FROM psc WHERE Name=:name",
+                ['name' => $site]
             );
-        }
 
-        $response = $uploader->handle($request);
-        if (!in_array($response->getStatusCode(), [200, 201], true)) {
-            // Something went wrong. Return early to skip further processing.
-            return $response;
-        }
+            // Set empty values equal to NULL for insertion into the database.
+            $instrument = !empty($body['instrument']) ? $body['instrument'] : null;
+            $pscid      = !empty($body['pscid'])      ? $body['pscid']      : null;
+            $visit      = !empty($body['visitLabel']) ? $body['visitLabel'] : null;
+            $comments   = !empty($body['comments'])   ? $body['comments']   : null;
+            $version    = !empty($body['version'])    ? $body['version']    : null;
 
-        // Prepare fields for database entry once upload has completed.
-        $site = $body['forSite']       !== '' ? $body['forSite'] : null;
-        $site = $DB->pselectOne(
-            "SELECT CenterID FROM psc WHERE Name=:name",
-            ['name' => $site]
-        );
+            $fileSize = $uploadedFile->getSize();
+            $fileType = pathinfo($filename, PATHINFO_EXTENSION);
 
-        // Set empty values equal to NULL for insertion into the database.
-        $instrument = !empty($body['instrument']) ? $body['instrument'] : null;
-        $pscid      = !empty($body['pscid'])      ? $body['pscid']      : null;
-        $visit      = !empty($body['visitLabel']) ? $body['visitLabel'] : null;
-        $comments   = !empty($body['comments'])   ? $body['comments']   : null;
-        $version    = !empty($body['version'])    ? $body['version']    : null;
+            // Insert record of this file into the `document_repository` table.
+            $DB->unsafeInsertOnDuplicateUpdate(
+                'document_repository',
+                [
+                    'File_category' => $category,
+                    'For_site'      => $site,
+                    'comments'      => $comments,
+                    'version'       => $version,
+                    'File_name'     => $filename,
+                    'File_size'     => $fileSize,
+                    'Data_dir'      => "$name/$filename",
+                    'uploaded_by'   => $name,
+                    'Instrument'    => $instrument,
+                    'PSCID'         => $pscid,
+                    'visitLabel'    => $visit,
+                    'File_type'     => $fileType,
+                ]
+            );
 
-        $fileSize = $uploadedFile->getSize();
-        $fileType = pathinfo($filename, PATHINFO_EXTENSION);
+            $uploadNotifier     = new \NDB_Notifier(
```

Comment:
This being in a loop means that the users will get one notification per file instead of one notification per upload action. Is that what we want? @ridz1208

Reply:
Very good observation. That was actually not a conscious intention on my part and can make the change if needed.

Reply:
The change has been made